shell/browser/api/electron_api_session.cc

```diff
@@ -150,6 +154,42 @@ uint32_t GetQuotaMask(const std::vector<std::string>& quota_types) {
   return quota_mask;
 }
 
+constexpr content::BrowsingDataRemover::DataType kClearDataTypeAll =
+    0xffffffffffffffffull;
+constexpr content::BrowsingDataRemover::OriginType kClearOriginTypeAll =
+    content::BrowsingDataRemover::ORIGIN_TYPE_UNPROTECTED_WEB |
+    content::BrowsingDataRemover::ORIGIN_TYPE_PROTECTED_WEB;
+
+// Observes the BrowsingDataRemover that backs the `clearBrowsingData` method
+// and resolves/rejects that API's promise once its done
+class ClearBrowsingDataObserver
+    : public content::BrowsingDataRemover::Observer {
+ public:
+  ClearBrowsingDataObserver(gin_helper::Promise<void> promise,
+                            content::BrowsingDataRemover* remover)
+      : promise_(std::move(promise)) {
+    observation_.Observe(remover);
+  }
+
+  void OnBrowsingDataRemoverDone(
+      content::BrowsingDataRemover::DataType failed_data_types) override {
+    if (failed_data_types == 0) {
```

Comment:
_(micronit)_ ^ if we had `-Wsigned` turned on, this would warn about comparison of signed & unsigned types

```suggestion
    if (failed_data_types == DataType{}) {
```

Reply:
I've changed this to `0ULL` for the following reasons:
* It should theoretically resolve the warning (didn't check though!)
* Since the `DataType` type must contain the full namespace, this will be quite verbose
* IMO it is clearer that `0` indicates "no flags" whereas `content::BrowsingDataRemover::DataType{}` doesn't read so clearly the same without spending some time analyzing the code and types involved, but I'm open to pushback on this opinion 😄